src/database/database.ts

```diff
@@ -85,3 +110,122 @@ export class DatabaseService implements FirebaseServiceInterface {
     });
   }
 }
+
+const RULES_URL_PATH = '.settings/rules.json';
+
+/**
+ * A helper client for managing RTDB security rules.
+ */
+class DatabaseRulesClient {
+
+  private readonly httpClient: AuthorizedHttpClient;
+  private readonly dbUrl: string;
+
+  constructor(app: FirebaseApp, dbUrl: string) {
+    this.httpClient = new AuthorizedHttpClient(app);
+    this.dbUrl = dbUrl;
+  }
+
+  /**
+   * Gets the currently applied security rules as a string. The return value consists of
+   * the rules source including comments.
+   *
+   * @return {Promise<string>} A promise fulfilled with the rules as a raw string.
+   */
+  public getRules(): Promise<string> {
+    const req: HttpRequestConfig = {
+      method: 'GET',
+      url: this.getRulesUrl(),
+    };
+    return this.httpClient.send(req)
+      .then((resp) => {
+        return resp.text;
+      })
+      .catch((err) => {
+        throw this.handleError(err);
+      });
+  }
+
+  /**
+   * Gets the currently applied security rules as a parsed JSON object. Any comments in
+   * the original source are stripped away.
+   *
+   * @return {Promise<object>} A promise fulfilled with the parsed rules source.
+   */
+  public getRulesJSON(): Promise<object> {
+    const req: HttpRequestConfig = {
+      method: 'GET',
+      url: `${this.getRulesUrl()}?format=strict`,
```

Comment:
should we be doing some kind of `resolve(...)` thing here? What happens if the dbUrl looks like `s-usc1c-nss-200.firebaseio.com?ns=foo`

Reply:
The URL returned by `getRulesUrl()` is guaranteed to end with `/.settings/rules.json`. If the original dbUrl had any query params, they do get stripped away by `getRulesUrl()`. Is that something we need to be concerned of? Does the user-specified DB URL ever have any query params?

Reply:
It can (notably true for the emulator, where it looks like `http://localhost:9000?ns=dbname`).

Reply:
Fixed, and added test cases.